lib/index.d.ts

```diff
@@ -0,0 +1,528 @@
+     */
+    nonce?: string
+  }
+
+  export interface ICallbackExtras {
+    /**
+     * extra request body properties to be sent to the AS during code exchange.
+     */
+    exchangeBody?: object
+    /**
+     * extra client assertion payload parameters to be sent as part of a client JWT assertion. This is only used
+     * when the client's token_endpoint_auth_method is either client_secret_jwt or private_key_jwt.
+     */
+    clientAssertionPayload?: object
+  }
+
+  export interface IRefreshExtras {
+    /**
+     * extra request body properties to be sent to the AS during refresh token exchange.
+     */
+    exchangeBody?: object
+    /**
+     * extra client assertion payload parameters to be sent as part of a client JWT assertion.
+     * This is only used when the client's token_endpoint_auth_method is either client_secret_jwt or private_key_jwt.
+     */
+    clientAssertionPayload?: object
+  }
+
+  export interface IGrantBody {
+    grant_type: string
+    [key: string]: any
+  }
+
+  export interface IGrantExtras {
+    /**
+     * extra client assertion payload parameters to be sent as part of a client JWT assertion.
+     * This is only used when the client's token_endpoint_auth_method is either client_secret_jwt or private_key_jwt.
+     */
+    clientAssertionPayload?: object
+  }
+
+  export interface IIntrospectExtras {
+    /**
+     * extra request body properties to be sent to the introspection endpoint.
+     */
+    introspectBody?: object
+    /**
+     * extra client assertion payload parameters to be sent as part of a client JWT assertion.
+     * This is only used when the client's token_endpoint_auth_method is either client_secret_jwt or private_key_jwt.
+     */
+    clientAssertionPayload: object
+  }
+
+  export interface IRevokeExtras {
+    /**
+     * extra request body properties to be sent to the revocation endpoint.
+     */
+    revokeBody?: object
+    /**
+     * extra client assertion payload parameters to be sent as part of a client JWT assertion.
+     * This is only used when the client's token_endpoint_auth_method is either client_secret_jwt or private_key_jwt.
+     */
+    clientAssertionPayload?: object
+  }
+
+  export interface IRequestObjectPayload {
+    client_id?: string
+    iss?: string
+    aud?: string
+    iat?: number
+    exp?: number
+    jti?: string
+    [key: string]: any
+  }
+
+  export interface IRegisterOther {
+    /**
+     * JWK Set formatted object with private keys used for signing client assertions or decrypting responses.
+     * When neither jwks_uri or jwks is present in metadata the key's public parts will be registered as jwks.
+     */
+    jwks?: object
+    /**
+     * Initial Access Token to use as a Bearer token during the registration call.
+     */
+    initialAccessToken?: string
+  }
+
+  export interface IDeviceAuthParameters {
+    client_id?: string
+    scope?: string
+    [key: string]: any
```

Comment:
`unknown` is a safer type to use here, so the developer has to explicitly cast it to what they believe the type should be